db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1430,10 +1432,8 @@ void DBImpl::NotifyOnCompactionCompleted(
   if (shutting_down_.load(std::memory_order_acquire)) {
     return;
   }
-  // TODO: Should disabling manual compaction squash compaction completed
-  //   notifications that aren't the result of a shutdown?
-  if (c->is_manual_compaction() &&
-      manual_compaction_paused_.load(std::memory_order_acquire) > 0) {
+
+  if (c->ShouldNotifyOnCompactionCompleted() == false) {
```

Comment:
Is this ever the case? Or is this left over from when I (incorrectly) claimed there is a bug where bottom-pri compactions see an extra `OnCompactionCompleted()`?

Reply:
Never mind this feels like Case II in the tests.